doc/api/n-api.md

```diff
@@ -2273,6 +2273,28 @@ Returns `napi_ok` if the API succeeded.
 This API checks if the Object passed in has the named property.
 
 
+#### *napi_delete_property*
+<!-- YAML
+added: REPLACEME
+-->
+```C
+napi_status napi_delete_property(napi_env env,
+                                 napi_value object,
+                                 napi_value key,
+                                 bool* result);
+```
+
+- `[in] env`: The environment that the N-API call is invoked under.
+- `[in] object`: The object to query.
+- `[in] key`: The name of the property to delete.
```

Comment:
I found [this section in the documentation](https://nodejs.org/api/n-api.html#n_api_working_with_javascript_properties) that provides some blanket statements for all of the methods for working with properties. Specifically, see:

> JavaScript value: these are represented in N-API by napi_value. This can be a napi_value representing a String, Number or Symbol.

Given that, and the surrounding documentation, I'd prefer to keep this as is in this PR. However, I think the n-api documentation as a whole should be made more consistent with the rest of the Node docs. For example, I think each method should document that `key` can be a number, string, or symbol. That way, users don't have to read the entire page/section/whatever to get that information.